### ISSUE TITLE
Add support for dictionaries and lists in attributes

### ISSUE THREAD

**Issue by mekaneck** [danieldotnl/ha-multiscrape#235]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to access parts of a single attribute within HA

**Describe the solution you'd like**
I'd like the attributes of multiscrape sensors to support dictionaries and lists. Currently the data shows up as text. 

**Describe alternatives you've considered**
Separate sensors for each dictionary entry

**Additional context**
The following two sensors are created with attribute data which should consist of dictionary and list data:

```yaml
multiscrape:
  - name: Multiscrape Date Test
    resource_template: http://date.jsontest.com
    scan_interval: 36000
    sensor:
      - unique_id: multiscrape_date_test
        name: Multiscrape Date Test
        value_template: "{{ value_json.start_date }}"
        attributes:
          - name: test
            value_template: >
              {% set myList = [1,2,3, "4th list item"] %}
              {{ {"key1": "value1", "key2": "value2", "keylist": myList} }}
template:
  - sensor:
      - name: Template Date Test
        unique_id: template_date_test
        state: "ok"
        attributes:
          test: >
            {% set myList = [1,2,3, "4th list item"] %}
            {{ {"key1": "value1", "key2": "value2", "keylist": myList} }}
```
In developer tools:
`{{ state_attr('sensor.multiscrape_date_test','test').keylist[3] }}` results in an error `UndefinedError: 'str object' has no attribute 'keylist'`
But yet:
`{{ state_attr('sensor.template_date_test','test').keylist[3] }}` results in `4th list item`

**Comment by danieldotnl** [comment]
This was requested in the past but I believe it is a limitation of HA. All attributes are treated as strings. Check the community forum please. If you find an integration that successfully keeps dictionaries/lists in the attributes, then let me know!

**Comment by mekaneck** [comment]
The google drive backup integration ([link](https://github.com/sabeechen/hassio-google-drive-backup)) provides a `sensor.backup_state` which has a list of dictionaries for the 'backups' attribute:
![image](https://github.com/danieldotnl/ha-multiscrape/assets/48502913/6b84815f-910a-4130-92f2-902bfd4f6b29)

I didn't have any luck finding other integrations with nested dicts/lists in an attribute for entities in the sensor domain. But the weather integrations commonly use this for forecasts for entities in the weather domain. Here's a weather entity from the [openweathermap integration](https://github.com/home-assistant/core/tree/dev/homeassistant/components/openweathermap):
![image](https://github.com/danieldotnl/ha-multiscrape/assets/48502913/a16cea88-5e40-43ab-9ccd-b03d85d38810)

The information I got from a mod in the HA Discord server was: 

> 1\) attributes are a dict, 2) anything JSON serializable can be assigned as the value to any key in that dict, and 3) dicts and lists are themselves JSON serializable

**Comment by danieldotnl** [comment]
I found it! Fixed it a temporary way by (ab)using a ha internal function. Created a PR to have this fixed in HA core.
Fixed in v6.7.3

**Comment by mekaneck** [comment]
Wow, great work! Thank you!!
